components/script/script_thread.rs

```diff
@@ -3318,75 +3277,12 @@ impl ScriptThread {
 
     /// Queue compositor events for later dispatching as part of a
     /// `update_the_rendering` task.
-    fn handle_event(&self, pipeline_id: PipelineId, event: CompositorEvent) {
+    fn handle_input_event(&self, pipeline_id: PipelineId, event: ConstellationInputEvent) {
         let Some(document) = self.documents.borrow().find_document(pipeline_id) else {
             warn!("Compositor event sent to closed pipeline {pipeline_id}.");
             return;
         };
-        document.note_pending_compositor_event(event);
-    }
-
-    #[allow(clippy::too_many_arguments)]
-    fn handle_mouse_button_event(
-        &self,
-        pipeline_id: PipelineId,
-        mouse_event_type: MouseEventType,
-        button: MouseButton,
-        point: Point2D<f32>,
-        node_address: Option<UntrustedNodeAddress>,
-        point_in_node: Option<Point2D<f32>>,
-        pressed_mouse_buttons: u16,
-        can_gc: CanGc,
-    ) {
-        let Some(document) = self.documents.borrow().find_document(pipeline_id) else {
-            warn!("Message sent to closed pipeline {pipeline_id}.");
-            return;
-        };
-        unsafe {
-            document.handle_mouse_button_event(
-                button,
-                point,
-                mouse_event_type,
-                node_address,
-                point_in_node,
-                pressed_mouse_buttons,
-                can_gc,
-            )
-        }
-    }
```

Comment:
These methods have been moved inline and now call the `Document` methods directly. That avoids an extra lookup of `Document` and the `unsafe` bits are just in a single line now.